File: swift/Programming/Programming/functional-programming/Closure.swift
//
//  Closure.swift
//  Programming
//
//  Created by ke on 2024/5/14.
//

import Foundation

class Closure {
    func closureExample() {
        let greet: (String) -> String = { name in
            return "Hello, \(name)!"
        }

        // Call the closure and assign its return value to a variable
        let greeting = greet("John")

        // Print the greeting
        print(greeting)
    }
}

File: go/closure/closure.go
package closure

import "fmt"

func closureExample() {
	greet := func(name string) string {
		return "Hello, " + name + "!"
	}

	// Call the function and assign its return value to a variable
	greeting := greet("John")

	// Print the greeting
	fmt.Println(greeting)
}
